manual/source/why.rst

```diff
@@ -0,0 +1,181 @@
+hence the word "sub-set" in our definition.
+
+Route Server
+~~~~~~~~~~~~
+
+A Route Server (as mentioned in RFC7947) would be a clear cut example
+of a BGP function. A Route Server requires the 'speaking' features of the
+BGP protocol and the BGP state machine, but it does not require the best-path
+selection mechanism, at least not in the form mentioned in RFC4271 and its
+updates.
+
+Route Reflector
+~~~~~~~~~~~~~~~
+
+Likewise, a Route Reflector (RFC4456) serves a specific function in an iBGP
+network. Again, it requires the 'speaking' features fo the BGP protocol, but
+it doesn't have to engage necessarily in best-path selection. Very simple
+Route Reflectors would not have a need for RIBS, they would just reflect the
+announcements and withdrawals they receive to its iBGP peers.
+
+Route Collector
+~~~~~~~~~~~~~~~
+
+A Route Collector is, broadly speaking, a device that passively engages in BGP
+sessions with the purpose of storing the learned routes together with
+meta-data about the whereabouts of these routes. Some of the purposes of
+storing these routes would be troubleshooting, and (longitudinal) analysis.
+
+"Engaging passively" may mean that the collector wil connect over BMP (BGP
+Monitoring Protocol), out-of-band, with one or more BGP routers. In this case
+the collector, called 'BMP Station' (RFC7854), will **not** be a node in the
+BGP network. It will only requires the packet parsing features of BGP in order
+to be able to extract the routes, and to be able to gather meta-data. 
+
+More commonly, though, Route Collectors **are** a node in the BGP network and
+the collector tries to "speak passively". As we saw, though, a passive speaker
+will have to engage in a minimum of speaking BGP. A Route Collector must never
+engage in best-path selection, or propagating routes to its peers whatsoever.
+Therefore Route Collectors have no need for keeping state in RIBs as described
+in RFC4271. Indeed, Route Collectors may completely forego having RIBs, since
+they can be synthesized later from the stored routes and meta-data if need be.
+
+Route Monitor
+~~~~~~~~~~~~~
+
+A Route Monitor is like a Route Collector in that it engages passively with
+BGP speakers through BMP or "passive speaking". However, instead of or in
+addition to storing, it will send signals to other systems ad/or applications
+based on certain user-defined events or combinations of (accumulated) events
+occurring in the observed BGP network. Some purposes would be troubleshooting,
+post-mortem analysis and anomaly detection.
+
+Off-line Looking Glass
+~~~~~~~~~~~~~~~~~~~~~~
+
+A looking glass is a command-line interface to a BGP speaker, quite often
+publicly available, that allows the user to query the content of the RIBs
+together with a few metrics. Many networks used to offer a web-interface that
+offers data directly to these looking glasses, but security as well as
+availability concerns led to unwillingness to continue doing this. Instead BGP
```

Comment:
So why does https://lg.twelve99.net/ exist?

Reply:
You're right, but this paragraph about LG is taking up too much space as it is (I already shortened it quite a bit), I am reluctant to elaborate. Maybe I'll just shove this whole paragraph into 'other functions'.